### PR TITLE
Draw New Recipe Window

### PR DESCRIPTION
**Pull request by cydanil**
This aims to fix the drawing of the new recipe window.

You should be able now to click `New` in the main window, and this should pop up. Not all functionalities are working though (such as the `yield` type or `cuisine` list.)

new_recipe_window_draw

### CODE REVIEW

File: gourmet/Undo.py [hunk 1]
```diff
@@ -80,7 +80,7 @@ class UndoableTextChange (UndoableObject):
     def __init__ (self, set_text_action, history, initial_text="",text="",txt_id=None,is_undo=False):
         self.txt_id = txt_id
         self.blob_matcher = re.compile('\s+\S+\s+')
-        self.initial_text = initial_text
+        self.initial_text = initial_text if initial_text is not None else ""
```
Comment (kirienko) [inline gourmet/Undo.py:92]: Doesn't `initial_text if initial_text else ""` produce the same result?

**Review by kirienko — APPROVED** [review]

File: gourmet/Undo.py (hunk 1 above)
Comment (cydanil) [reply gourmet/Undo.py:92]: In that context, yes. However, I find that this formulation is more explicit: there's someone somewhere handing us `None`, and we're explicitly avoiding that value here, as it ruins the `undo` functionality later.

File: gourmet/Undo.py [hunk 2]
```diff
@@ -92,7 +92,7 @@ def __init__ (self, set_text_action, history, initial_text="",text="",txt_id=Non
             debug('Too many changes - assume 0,0',0)
             self.cindex,self.clen = 0,0
 
-    def determine_mode (self,text=None,initial_text=None):
+    def determine_mode(self, text="", initial_text="") -> str:
```
Comment (maweki) [inline gourmet/Undo.py]: can we have input types as well? :)

Comment (cydanil) [reply gourmet/Undo.py]: yep!

**Comment by cydanil** [comment]
Note that I've added a commit, handling Gtk.ComboBoxText in the reciped editor module, meaning that we can choose from a drop-down menu correctly. I don't feel confident at the moment to refactor the whole `if`, though.

So the `category`, `cuisine`, and `source` fields can be populated as well.

In another PR, I'll fix the ingredients input.

File: gourmet/Undo.py [hunk 3]
```diff
@@ -92,7 +92,7 @@ def __init__ (self, set_text_action, history, initial_text="",text="",txt_id=Non
             debug('Too many changes - assume 0,0',0)
             self.cindex,self.clen = 0,0
 
-    def determine_mode(self, text="", initial_text="") -> str:
+    def determine_mode(self, text: str = "", initial_text: str = "") -> str:
```
Comment (maweki) [inline gourmet/Undo.py]: If both arguments can be called with `None` (next two lines) it should be `Optional[str]`. Or do we actually check for empty string? What's the semantic difference between both?

Comment (cydanil) [reply gourmet/Undo.py]: This should indeed be marked as `Optional[str]`

Comment (cydanil) [reply gourmet/Undo.py]: This should indeed be marked as `Optional[str]`

Comment (cydanil) [reply gourmet/Undo.py]: Done

File: gourmet/Undo.py [hunk 4]
```diff
@@ -92,7 +94,9 @@ def __init__ (self, set_text_action, history, initial_text="",text="",txt_id=Non
             debug('Too many changes - assume 0,0',0)
             self.cindex,self.clen = 0,0
 
-    def determine_mode(self, text: str = "", initial_text: str = "") -> str:
+    def determine_mode(self,
+                       text: Optional[str] = "",
+                       initial_text: Optional[str] = "") -> str:
```
Comment (maweki) [inline gourmet/Undo.py]: I don't want to be a burden, but we only return `add` and `delete` and we don't have to change this here and now (we can create a new ticket and PR for this), but this looks suspiciously like add/delete it should be an `Enum`.

Comment (maweki) [reply gourmet/Undo.py]: otherwise I am fine with this. We just have to note this for later.

Comment (cydanil) [reply gourmet/Undo.py]: Maybe a boolean?

Comment (maweki) [reply gourmet/Undo.py]: Definitely an Enum. Boolean is also an Enum but True/False is not a "Mode".

Comment (cydanil) [reply gourmet/Undo.py]: Ok, that did uncover a bug. Can you have at it again?

**Comment by maweki** [comment]
Cool. I was proposing changing that in a different PR because now we have to chase down usages.

https://github.com/kirienko/gourmet/blob/f2d13b16d8d96d852379f5f6ada65347a18c4c46/gourmet/Undo.py#L132-L135

https://github.com/kirienko/gourmet/blob/f2d13b16d8d96d852379f5f6ada65347a18c4c46/gourmet/Undo.py#L147-L150

**Comment by cydanil** [comment]
I changed the enums back to string, because of the `if` statement line 116 :
https://github.com/kirienko/gourmet/blob/f2d13b16d8d96d852379f5f6ada65347a18c4c46/gourmet/Undo.py#L116-L118

**Comment by maweki** [comment]
> I changed the enums back to string, because of the `if` statement line 116 :
> 
> https://github.com/kirienko/gourmet/blob/f2d13b16d8d96d852379f5f6ada65347a18c4c46/gourmet/Undo.py#L116-L118

Okay. We do that later. The lines you've shown should never be hit because of

https://github.com/kirienko/gourmet/blob/f2d13b16d8d96d852379f5f6ada65347a18c4c46/gourmet/Undo.py#L97

 So the `mode` property is `UndoMode` and the `UndoMode` is, as far as I have seen, `add` and `delete`. I think extracting that would make everything more readable. But let's not do that here.

**Comment by cydanil** [comment]
Yes, I had noticed as well, but it's too out of the scope here, as you noted.

**Pull request merged**